app/controllers/api/vms_controller.rb

```diff
@@ -351,7 +351,7 @@ def vm_event(vm, event_type, event_message, event_time)
     def retire_vm(vm, id, data)
       desc = "#{vm_ident(vm)} retiring"
       desc << " on #{data['date']}" if Hash(data)['date'].present?
-      retire_resource(:vms, id, data)
+      method(:retire_resource).super_method.call(:vms, id, data)
```

Comment:
Can you explain why this is necessary?  Wouldn't you still be able to call retire_resource as before?

If not, perhaps it would feel less clunky to alias the parent's method (perhaps to `retire_resource_vms`) and then call that alias directly?